internal/controller/dispatch_logic.go

```diff
@@ -61,13 +63,42 @@ LOOP:
 		}
 		for _, pod := range pods.Items {
 			if _, ok := pod.GetLabels()[nameLabel]; !ok && pod.Status.Phase != v1.PodFailed && pod.Status.Phase != v1.PodSucceeded {
-				capacity.Sub(NewWeightsForPod(&pod))
+				nodeCapacity.Sub(NewWeightsForPod(&pod))
 			}
 		}
+		// add allocatable capacity on the node
+		capacity.Add(nodeCapacity)
+		updateMetrics(capacity, node)
 	}
 	return capacity, nil
 }
 
+// Update metrics
+func updateMetrics(capacity Weights, node v1.Node) {
+	// In general, it's not recommended to convert Dec to float64. However, we have no choice since Prometheus works with float64.
+	// https://github.com/go-inf/inf/issues/7#issuecomment-504729949
+	capacityCpu, err := strconv.ParseFloat(capacity["cpu"].String(), 64)
+	if err != nil {
+		mcadLog.V(1).Error(err, "Unable to get CPU capacity", "node", node.Name)
+	} else {
+		totalCapacityCpu.WithLabelValues(node.Name).Set(capacityCpu)
+	}
+
+	capacityMemory, err := strconv.ParseFloat(capacity["memory"].String(), 64)
+	if err != nil {
+		mcadLog.V(1).Error(err, "Unable to get memory capacity", "node", node.Name)
+	} else {
+		totalCapacityMemory.WithLabelValues(node.Name).Set(capacityMemory)
+	}
+
+	capacityGpu, err := strconv.ParseFloat(capacity["nvidia.com/gpu"].String(), 64)
+	if err != nil {
+		mcadLog.V(1).Error(err, "Unable to get gpu capacity", "node", node.Name)
```

Comment:
It appears that error log entries are always being logged (regardless of their verbosity level).
https://github.com/go-logr/logr#background

On GPU-less clusters, this line will fill up the log as it is written priodically for every node. Should I change it to `Info()`? Should I change to `Info()` the other error mesages in this function as well?

Reply:
I suggest you first check that capacity["nvidia.com/gpu"] is defined and only attempt the conversion to float64 if is (or default capacity["nvidia.com/gpu"] to 0 in place nil). The error messages can/should remain  errors. They should not be reachable.

Reply:
Let's not introduce verbosity levels in the code yet (`V(1)`), at least until we have clear, agreed upon logic for it.

Reply:
Sure. makes sense. done